share/spack/setup-env.sh

```diff
@@ -315,6 +311,21 @@ if ! _spack_fn_exists use && ! _spack_fn_exists module; then
 	need_module="yes"
 fi;
 
+# Define the spack shell function with some informative no-ops, so when users
+# run `which spack`, they see the path to spack and where the function is from.
+eval "spack() {
```

Comment:
LGTM.  

If this a *real* lack of coverage issue, please add a test for the `eval` here to get the patch coverage up.